pkg/v1/providers/infrastructure-byoh/v0.1.0/ytt/overlay.yaml

```diff
@@ -0,0 +1,191 @@
+    name: kube-vip
+    resources: {}
+    securityContext:
+      capabilities:
+        add:
+        - NET_ADMIN
+        - SYS_TIME
+    volumeMounts:
+    - mountPath: /etc/kubernetes/admin.conf
+      name: kubeconfig
+  hostNetwork: true
+  volumes:
+  - hostPath:
+      path: /etc/kubernetes/admin.conf
+      type: FileOrCreate
+    name: kubeconfig
+status: {}
+#@ end
+
+
+
+
+#@overlay/match by=overlay.subset({"kind":"Cluster"})
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  #@ if data.values.BYOH_CONTROL_PLANE_ENDPOINT:
+  #@overlay/match missing_ok=True
+  annotations:
+    tkg.tanzu.vmware.com/cluster-controlplane-endpoint: #@ data.values.BYOH_CONTROL_PLANE_ENDPOINT
+  #@ end
+  labels:
+    #@overlay/match missing_ok=True
+    #@yaml/text-templated-strings
+    #@ if data.values.TKG_CLUSTER_ROLE != "workload":
+    cluster-role.tkg.tanzu.vmware.com/(@= data.values.TKG_CLUSTER_ROLE @): ""
+    #@ end
+    tkg.tanzu.vmware.com/cluster-name: #@ data.values.CLUSTER_NAME
+    #@overlay/match missing_ok=True
+    tanzuKubernetesRelease: #@ data.values.KUBERNETES_RELEASE
+spec:
+  infrastructureRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: ByoCluster
+    name: #@ data.values.CLUSTER_NAME
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: KubeadmControlPlane
+    name: #@ "{}-control-plane".format(data.values.CLUSTER_NAME)
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      #@overlay/match by=overlay.index(0)
+      - #@ data.values.CLUSTER_CIDR
+    services:
+      cidrBlocks:
+      #@overlay/match by=overlay.index(0)
+      - #@ data.values.SERVICE_CIDR
+
+#@overlay/match by=overlay.subset({"kind":"ByoCluster"})
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: ByoCluster
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+spec:
+  #@ if not data.values.AVI_CONTROL_PLANE_HA_PROVIDER:
+  controlPlaneEndpoint:
+    host: #@ data.values.BYOH_CONTROL_PLANE_ENDPOINT
+    port: 6443
+  #@ else:
+  #@overlay/remove
+  controlPlaneEndpoint:
+  #@ end  
+
+
+#@overlay/match by=overlay.subset({"kind": "ByoMachineTemplate", "metadata":{"name": "${ CLUSTER_NAME }-control-plane"}})
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: ByoMachineTemplate
+metadata:
+  name: #@ "{}-control-plane".format(data.values.CLUSTER_NAME)
+spec:
+  template:
+    spec: {}
+
+#@overlay/match by=overlay.subset({"kind":"KubeadmControlPlane"})
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  name: #@ "{}-control-plane".format(data.values.CLUSTER_NAME)
+spec:
+  kubeadmConfigSpec:      
+    files:
+    #@ if not data.values.AVI_CONTROL_PLANE_HA_PROVIDER:
+    #@overlay/match by=overlay.index(0)
+    - content: #@ yaml.encode(kube_vip_pod())
+    #@ else:
+    #@overlay/match by=overlay.index(0)  
+    #@overlay/remove
+    - content:
+    #@ end
+    users:
```

Comment:
BYOH don't need this section.   Please update this file according to the cluster template file.